py-polars/polars/internals/frame.py

```diff
@@ -1638,6 +1652,59 @@ def _pos_idx(self, idx: int, dim: int) -> int:
         else:
             return self.shape[dim] + idx
 
+    def _pos_idxs(self, idxs: np.ndarray | pli.Series, dim: int) -> pli.Series:
+        if isinstance(idxs, pli.Series):
+            if idxs.dtype == UInt32:
+                return idxs
+            if idxs.dtype in {
+                UInt8,
+                UInt16,
+                UInt64,
+                Int8,
+                Int16,
+                Int32,
+                Int64,
+            }:
+                if idxs.dtype in {Int64, UInt64}:
+                    if idxs.max() >= 2**32:  # type: ignore[operator]
+                        raise ValueError("Index positions should be smaller than 2^32.")
+                if idxs.dtype == Int64:
+                    if idxs.min() < -(2**32):  # type: ignore[operator]
```

Comment:
This is not true in polars-big-idx.

Reply:
Yeah I know it is not true. I think it would be good to have a proper way to detect the idx width. a pl.IDX datatype could maybe help too.